### PR TITLE
feat: HMR for changing styles injected into shadow roots

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

File: flow-tests/test-live-reload/frontend/component-with-theme.ts [hunk 1]
```diff
@@ -16,6 +18,9 @@ export class ComponentWithTheme extends LitElement {
   }
   connectedCallback() {
     super.connectedCallback();
-    applyTheme(this.renderRoot);
+    if (!this.applied) {
+      applyTheme(this.renderRoot);
+      this.applied = true;
+    }
```
Comment (caalador) [inline flow-tests/test-live-reload/frontend/component-with-theme.ts:24]: Does this have an implication that all components should check applyTheme usage so it is only called once?
If so, could applyTheme have book keeping on the applied elements.

Comment (Artur-) [reply flow-tests/test-live-reload/frontend/component-with-theme.ts:24]: Maybe this should be in `applyTheme` instead. But has anything changed here? Didn't the old code apply the theme multiple times?

File: flow-server/src/main/resources/META-INF/frontend/theme-util.js [hunk 2]
```diff
@@ -81,9 +88,9 @@ export const injectGlobalCss = (css, referenceComment, target, first) => {
 
   // We avoid mixing style tags and adoptedStyleSheets to make override order clear
   if (target === document) {
-    addStyleTag(cssText, referenceComment);
+    return addStyleTag(cssText, referenceComment);
   } else {
-    addAdoptedStyle(cssText, target, first);
+    return addAdoptedStyle(cssText, target, first);
   }
```
[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/resources/META-INF/frontend/theme-util.js]: Fixed

File: flow-tests/test-live-reload/frontend/component-with-theme.ts (hunk 1 above)
Comment (Artur-) [reply flow-tests/test-live-reload/frontend/component-with-theme.ts]: I reverted this test change as it does not seem to be needed. If `applyTheme` applies twice when run twice, then it all the rules should also be removed with this approach, as all removers are collected into an array

So maybe it should be fixed but not in this PR, unless something breaks now

**Review by caalador — APPROVED** [review]

**Pull request merged**